ttl_test.go

```diff
@@ -174,6 +176,121 @@ func TestTTLScenario(t *testing.T) {
 	}
 }
 
+func TestHeadTailLogicConcurrent(t *testing.T) {
+	m := NewMapTTLCache[string, string](context.Background(), time.Millisecond, time.Hour)
```

Comment:
I usually use context with deferred cancel to avoid having live goroutine after test is finished (this sometimes causes test to fail, eg when using goleak).

Reply:
added